lib/iris/cube.py

```diff
@@ -3940,7 +3940,15 @@ def __next__(self):
 
         if self._ordered:
             if any(self._mod_requested_dims != list(range(len(cube.shape)))):
-                cube.transpose(self._mod_requested_dims)
+                sliced_dims = []
+                for i in self._requested_dims:
+                    # Convert dim numbers back to dims to match up to indices
+                    # of sliced cube
+                    dim_ref = self._cube.dim_coords[i].name()
```

Comment:
Dammit, I was hoping that this wouldn't happen using this approach.  I'll have to see if I can do anything about it, but I'm a little worried that I might have to discuss this in-depth with somebody to be able to work out the best strategy.